### ISSUE TITLE
`sf package install` provides incorrect errors with missing/incorrect installation key

### ISSUE THREAD

**Issue by jclark-dot-org** [forcedotcom/cli#2882]
### Summary
When using `sf package install` to install a package that was created with an installation key, if you omit the installation key or provide an incorrect key, the error message is misleading, unless you also include the `-b | --publish-wait` flag. 

Please note, this has only been tested with unlocked packages.

### Steps To Reproduce
You can reproduce this with any package that has an installation key; I've provided one of my unlocked package versions below.  Feel free to substitute any unlocked package version with an installation key.

Example 1:  Attempt to install the package without an installation key:
```
sf package install -w 30 -p 04tQi0000006nyxIAA -a package
```

Example 2:  Attempt to install the package with an incorrect installation key:
```
sf package install -w 30 -p 04tQi0000006nyxIAA -a package -k SEEKRIT
```

### Expected result
An appropriate error message should be displayed, such as:

Example 1:
> Please provide installationkey when querying against SubscriberPackageVersion that is protected by installationkey

Example 2:
> Invalid InstallationKey for SubscriberPackageVersion

### Actual result
Both examples fail with the error:
```
Error (1): The subscriber package version 04tQi0000006nyxIAA is invalid, no subscriber package version record found.
```
**NOTE**
If you add the `-b | --publish-wait` flag, the incorrect error is replaced with another incorrect error:
> The subscriber package version is not fully available.

BUT, it is then followed with the message "Try This:", which is followed by the correct, useful error messages from the Expected Results above.

### System Information

zsh (oh my zsh), on Iterm2 on Mac OS Sonoma

```json
{
  "architecture": "darwin-arm64",
  "cliVersion": "@salesforce/cli/2.41.8",
  "nodeVersion": "node-v21.4.0",
  "osVersion": "Darwin 23.3.0",
  "rootPath": "/Users/jclark/.nvm/versions/node/v21.4.0/lib/node_modules/@salesforce/cli",
  "shell": "zsh",
  "pluginVersions": [
    "@oclif/plugin-autocomplete 3.0.17 (core)",
    "@oclif/plugin-commands 3.3.2 (core)",
    "@oclif/plugin-help 6.0.21 (core)",
    "@oclif/plugin-not-found 3.1.8 (core)",
    "@oclif/plugin-plugins 5.0.19 (core)",
    "@oclif/plugin-search 1.0.23 (core)",
    "@oclif/plugin-update 4.2.11 (core)",
    "@oclif/plugin-version 2.0.18 (core)",
    "@oclif/plugin-warn-if-update-available 3.0.16 (core)",
    "@oclif/plugin-which 3.1.9 (core)",
    "@salesforce/cli 2.41.8 (core)",
    "apex 3.1.11 (core)",
    "auth 3.6.9 (core)",
    "data 3.3.4 (core)",
    "deploy-retrieve 3.6.13 (core)",
    "info 3.2.7 (core)",
    "limits 3.3.5 (core)",
    "marketplace 1.2.5 (core)",
    "org 4.1.8 (core)",
    "packaging 2.4.2 (core)",
    "schema 3.3.6 (core)",
    "settings 2.2.6 (core)",
    "sobject 1.4.1 (core)",
    "source 3.3.9 (core)",
    "telemetry 3.3.5 (core)",
    "templates 56.2.5 (core)",
    "trust 3.6.8 (core)",
    "user 3.5.5 (core)"
  ]
}

```

**Comment by jclark-dot-org** [comment]
I have updated to 2.42.6 and have confirmed that the issue remains; all outputs in the original issue remain unchanged.  

New version info:
```JSON
{
  "architecture": "darwin-arm64",
  "cliVersion": "@salesforce/cli/2.42.6",
  "nodeVersion": "node-v21.4.0",
  "osVersion": "Darwin 23.3.0",
  "rootPath": "/Users/jclark/.nvm/versions/node/v21.4.0/lib/node_modules/@salesforce/cli",
  "shell": "zsh",
  "pluginVersions": [
    "@oclif/plugin-autocomplete 3.0.18 (core)",
    "@oclif/plugin-commands 3.3.4 (core)",
    "@oclif/plugin-help 6.0.22 (core)",
    "@oclif/plugin-not-found 3.1.9 (core)",
    "@oclif/plugin-plugins 5.0.21 (core)",
    "@oclif/plugin-search 1.0.24 (core)",
    "@oclif/plugin-update 4.2.12 (core)",
    "@oclif/plugin-version 2.1.2 (core)",
    "@oclif/plugin-warn-if-update-available 3.0.19 (core)",
    "@oclif/plugin-which 3.1.10 (core)",
    "@salesforce/cli 2.42.6 (core)",
    "apex 3.1.11 (core)",
    "auth 3.6.11 (core)",
    "data 3.3.5 (core)",
    "deploy-retrieve 3.6.14 (core)",
    "info 3.2.8 (core)",
    "limits 3.3.7 (core)",
    "marketplace 1.2.5 (core)",
    "org 4.1.10 (core)",
    "packaging 2.4.4 (core)",
    "schema 3.3.6 (core)",
    "settings 2.2.6 (core)",
    "sobject 1.4.2 (core)",
    "source 3.3.12 (core)",
    "telemetry 3.3.7 (core)",
    "templates 56.2.7 (core)",
    "trust 3.6.11 (core)",
    "user 3.5.8 (core)"
  ]
}
```

**Comment by iowillhoit** [comment]
Thanks for the issue @jclark-dot-org, we will get this looked at and try to improve this messaging.
